machine_learning/loss_functions/hinge_loss.py

```diff
@@ -0,0 +1,55 @@
+"""
+Hinge Loss
+
+Description:
+Compute the Hinge loss used for training SVM (Support Vector Machine).
+
+Formula:
+loss = max(0, 1 - true * pred)
+
+Reference: https://en.wikipedia.org/wiki/Hinge_loss
+
+Author: Poojan Smart
+Email: smrtpoojan@gmail.com
+"""
+
+import numpy as np
+
+
+def hinge_loss(y_true: np.ndarray, pred: np.ndarray) -> float:
+    """
+    Calculate the hinge loss for y_true and pred for binary classification.
+
+    Args:
+        y_true: Array of actual values (ground truth) encoded as -1 and 1.
```

Comment:
Please have an if-statement enforce that `y_true` must be encoded as -1 or 1 (and please raise a ValueError for invalid inputs)